tests/bluetooth/tester/src/btp_gap.c

```diff
@@ -30,7 +30,8 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_BTTESTER_LOG_LEVEL);
 #define CONTROLLER_NAME "btp_tester"
 
 #define BT_LE_AD_DISCOV_MASK (BT_LE_AD_LIMITED | BT_LE_AD_GENERAL)
-#define ADV_BUF_LEN (sizeof(struct btp_gap_device_found_ev) + 2 * 31)
+/*consider ext_adv case*/
+#define ADV_BUF_LEN (sizeof(struct btp_gap_device_found_ev) + 2 * 229)
```

Comment:
Hmm I don't think we need  2*229 as for ext adv there is either AD or SCAN_RSP, never both.
Also, maybe this should be CONFIG_BT_EXT_SCAN_BUF_SIZE instead of hardcoded 229 ?


And this buffer is used to concatenate AD and SCAN_RSP in BTP event.

Reply:
Hi @sjanc   The macro CONFIG_BT_EXT_ADV is set yes in bt_tester project,  so when start scan, actually start le ext scan, 
the data len of scan rsp maybe  more than 31.

Reply:
Hi,

I mean that this could be
#define ADV_BUF_LEN (sizeof(struct btp_gap_device_found_ev) + CONFIG_BT_EXT_SCAN_BUF_SIZE);

or maybe to be sure we always fit full legacy AD+SCAN_RSP
#define ADV_BUF_LEN (sizeof(struct btp_gap_device_found_ev) + MAX(2 * 31, CONFIG_BT_EXT_SCAN_BUF_SIZE));

It should cover both cases since in ext adv PDUs it is either AD or SCAN_RSP, never both

Reply:
Hi @sjanc Thanks your comment, I have modified the code, can you help review again

Reply:
Hi @sjanc  Do you have any other suggestions on your end